ynvest_tube/app/src/main/java/com/app/ynvest_tube/refresher/DataRefresher.kt

```diff
@@ -0,0 +1,84 @@
+package com.app.ynvest_tube.refresher
+
+import android.content.Context
+import android.widget.TextView
+import android.widget.Toast
+import com.app.ynvest_tube.R
+import com.app.ynvest_tube.adapters.AuctionsAdapter
+import com.app.ynvest_tube.adapters.CurrentRentalsAdapter
+import com.app.ynvest_tube.adapters.PreviousRentalsAdapter
+import com.app.ynvest_tube.model.Auction
+import com.app.ynvest_tube.model.AuctionDetailsResponse
+import com.app.ynvest_tube.model.User
+import com.app.ynvest_tube.model.UserDetailsResponse
+import com.app.ynvest_tube.repository.Repository
+import java.lang.Thread.sleep
+
+class DataRefresher {
+    companion object {
+        private lateinit var repository: Repository
+
+        private lateinit var refreshTask: Thread
+
+        public var userSubscribers = mutableMapOf<String, UserSubscriber>()
+        public var userDetailsSubscribers = mutableMapOf<String, UserDetailsSubscriber>()
+        public var auctionListSubscribers = mutableMapOf<String, AuctionListSubscriber>()
+        public var auctionSubscribers = mutableMapOf<String, AuctionSubscriber>()
+
+        public var toastContext: Context? = null
```

Comment:
I know about possible memory leak here. It is about to be deeper researched/fixed